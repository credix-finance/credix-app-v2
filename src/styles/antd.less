@import "antd/dist/antd.less";

.ant-btn,
.ant-btn:active,
.ant-btn:focus,
.ant-btn:hover {
	// Disable the default antd button background so it doesn't interfere with Tailwind css bg-* classes.
	background: none;
}
