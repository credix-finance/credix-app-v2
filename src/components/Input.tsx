import React, { useContext } from "react";
import { Input as AntdInput } from "antd";
import { InputProps as AntInputProps } from "antd/lib/input";
import { InputContext } from "./Label";

interface InputProps {
	/**
	 * Controls wheter the input can be interacted with or not.
	 * This differs from setting `disabled` to `true` in that by
	 * setting this option to true will remove any styling from the
	 * input that makes it look like an input.
	 */
	isDisplay?: boolean;
	placeholder?: AntInputProps["placeholder"];
	onClick?: AntInputProps["onClick"];
	className?: AntInputProps["className"];
	children?: AntInputProps["children"];
	value?: AntInputProps["value"];
}

export const Input = ({
	children,
	value,
	className = "",
	isDisplay = false,
	...props
}: InputProps) => {
	const { hasFeedback, validateStatus, disabled } = useContext(InputContext);

	if (isDisplay) {
		return <div className="font-medium text-base pt-2 pb-4">{value}</div>;
	}

	return (
		<AntdInput
			disabled={disabled}
			value={value}
			className={`
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary placeholder-neutral-100/70 border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				${
					hasFeedback && validateStatus == "error"
						? "border-error focus:ring-error focus:border-error"
						: "border-neutral-60 focus:ring-neutral-100 focus:border-neutral-100 "
				}
				${className}
			`}
			{...props}
		>
			{children}
		</AntdInput>
	);
};
