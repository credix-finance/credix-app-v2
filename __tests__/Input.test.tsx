import React from "react";
import renderer from "react-test-renderer";
import { Input } from "@components/Input";
import { Form } from "antd";

const defaultArgs = {
	size: "middle",
	label: "Borrower Key",
	placeholder: "key",
	disabled: false,
	isDisplay: false,
	value: "USBDCKTVGBBK",
};

test("Default input", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input label={defaultArgs.label} placeholder={defaultArgs.placeholder} />
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Input with text", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input
				label={defaultArgs.label}
				placeholder={defaultArgs.placeholder}
				value={defaultArgs.value}
			/>
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Input with error feedback", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input
				label={defaultArgs.label}
				hasFeedback={true}
				validateStatus="error"
				placeholder={defaultArgs.placeholder}
				value={defaultArgs.value}
			/>
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Dispaly input", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input
				label={defaultArgs.label}
				placeholder={defaultArgs.placeholder}
				value={defaultArgs.value}
				isDisplay={true}
			/>
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Disabled input", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input
				label={defaultArgs.label}
				disabled={true}
				placeholder={defaultArgs.placeholder}
				value={defaultArgs.value}
			/>
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Input with description", () => {
	const component = renderer.create(
		<Form layout="vertical">
			<Input
				label={"This is a longer question with some more informations under it?"}
				description="This is a short information text that gives more details about the question."
				placeholder={defaultArgs.placeholder}
				value={defaultArgs.value}
			/>
		</Form>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
