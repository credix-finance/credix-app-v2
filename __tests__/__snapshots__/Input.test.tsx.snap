// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default input 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				undefined
			"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="Borrower Key"
      >
        Borrower Key
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input 
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary
				placeholder-neutral-100/70
				border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				border-neutral-60 focus:ring-neutral-100 focus:border-neutral-100 
				
			"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="key"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Disabled input 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				undefined
			"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="Borrower Key"
      >
        Borrower Key
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input ant-input-disabled 
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary
				placeholder-neutral-100/70
				border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				border-neutral-60 focus:ring-neutral-100 focus:border-neutral-100 
				
			"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="key"
            type="text"
            value="USBDCKTVGBBK"
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Dispaly input 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				undefined
			"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="Borrower Key"
      >
        Borrower Key
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="font-medium text-base pt-2 pb-4"
          >
            USBDCKTVGBBK
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Input with description 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				undefined
			"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="This is a longer question with some more informations under it?"
      >
        This is a longer question with some more informations under it?
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="font-normal text-sm mt-0 mb-[10px]"
          >
            This is a short information text that gives more details about the question.
          </div>
          <input
            className="ant-input ant-input-disabled 
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary
				placeholder-neutral-100/70
				border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				border-neutral-60 focus:ring-neutral-100 focus:border-neutral-100 
				
			"
            disabled={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="key"
            type="text"
            value="USBDCKTVGBBK"
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Input with error feedback 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				border-error
			 ant-form-item-has-feedback ant-form-item-has-error"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="Borrower Key"
      >
        Borrower Key
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input 
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary
				placeholder-neutral-100/70
				border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				border-error focus:ring-error focus:border-error
				
			"
            disabled={false}
            hasFeedback={true}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="key"
            type="text"
            validateStatus="error"
            value="USBDCKTVGBBK"
          />
        </div>
        <span
          className="ant-form-item-children-icon"
        >
          <span
            aria-label="close-circle"
            className="anticon anticon-close-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
              />
            </svg>
          </span>
        </span>
      </div>
    </div>
  </div>
</form>
`;

exports[`Input with text 1`] = `
<form
  className="ant-form ant-form-vertical"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item 
				font-bold text-base
				false
				undefined
			"
    style={Object {}}
  >
    <div
      className="ant-col ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        title="Borrower Key"
      >
        Borrower Key
      </label>
    </div>
    <div
      className="ant-col ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input 
				pl-4 pr-[25px] py-3 font-medium text-base bg-credix-primary
				placeholder-neutral-100/70
				border-[0.5px] rounded-[1px]
				focus:shadow-none
				disabled:border-neutral-60/40
				border-neutral-60 focus:ring-neutral-100 focus:border-neutral-100 
				
			"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="key"
            type="text"
            value="USBDCKTVGBBK"
          />
        </div>
      </div>
    </div>
  </div>
</form>
`;
